grakn-graql/src/main/java/ai/grakn/graql/internal/parser/QueryVisitor.java

```diff
@@ -717,17 +730,6 @@ private Order getOrder(TerminalNode order) {
     }
 
     private ResourceType.DataType getDatatype(TerminalNode datatype) {
-        switch (datatype.getText()) {
-            case "long":
-                return ResourceType.DataType.LONG;
-            case "double":
-                return ResourceType.DataType.DOUBLE;
-            case "string":
-                return ResourceType.DataType.STRING;
-            case "boolean":
-                return ResourceType.DataType.BOOLEAN;
-            default:
-                throw new RuntimeException("Unrecognized datatype " + datatype.getText());
-        }
+        return QueryParser.DATA_TYPES.get(datatype.getText());
```

Comment:
What will happen now when the datatype is not in the map? Is this prevented by ANTLR?